src/mscorlib/src/System/Environment.cs

```diff
@@ -1321,38 +1321,42 @@ internal static String GetStackTrace(Exception e, bool needFileInfo)
 
         internal static string GetResourceString(string key, object val0)
         {
-            return GetResourceString(key, new object[] { val0 });
+            return GetResourceString(new object[] { val0 }, key);
         }
 
         internal static string GetResourceString(string key, object val0, object val1)
         {
-            return GetResourceString(key, new object[] { val0, val1 });
+            return GetResourceString(new object[] { val0, val1 }, key);
         }
 
         internal static string GetResourceString(string key, object val0, object val1, object val2)
         {
-            return GetResourceString(key, new object[] { val0, val1, val2 });
+            return GetResourceString(new object[] { val0, val1, val2 }, key);
         }
 
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3)
         {
-            return GetResourceString(key, new object[] { val0, val1, val2, val3 });
+            return GetResourceString(new object[] { val0, val1, val2, val3 }, key);
         }
 
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3, object val4)
         {
-            return GetResourceString(key, new object[] { val0, val1, val2, val3, val4 });
+            return GetResourceString(new object[] { val0, val1, val2, val3, val4 }, key);
         }
 
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3, object val4, object val5)
         {
-            return GetResourceString(key, new object[] { val0, val1, val2, val3, val4, val5 });
+            return GetResourceString(new object[] { val0, val1, val2, val3, val4, val5 }, key);
         }
 
+        internal static String GetResourceString(string key, params object[] values)
+        {
+            return GetResourceString(values, key);
+        }
 
         // NoInlining causes the caller and callee to not be inlined in mscorlib as it is an assumption of StackCrawlMark use
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static String GetResourceString(string key, params object[] values)
+        private static String GetResourceString(object[] values, string key)
```

Comment:
Instead of shifting around all of the parameters which will make this confusing for readers + generates a larger git diff, you could just call this `GetResourceStringCore` or `GetResourceStringInternal`?